nominal/core/stream_v2.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import annotations
+
+import concurrent.futures
+import enum
+import logging
+import threading
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta
+from types import TracebackType
+from typing import Callable, Sequence, Type
+
+from typing_extensions import Self
+
+from nominal.core._clientsbunch import ProtoWriteService
+from nominal.core.queueing import (
+    BackpressureQueue,
+    BlockingQueue,
+    DropNewestQueue,
+    DropOldestQueue,
+    ReadQueue,
+    iter_queue,
+    spawn_batching_thread,
+)
+from nominal.core.stream import BatchItem
+
+# if TYPE_CHECKING:
+from nominal.core.worker_pool import ProcessPoolManager
+from nominal.ts import IntegralNanosecondsUTC
+
+logger = logging.getLogger(__name__)
+
+
+class BackpressureMode(enum.Enum):
+    BLOCK = "block"
+    DROP_NEWEST = "drop_newest"
+    DROP_OLDEST = "drop_oldest"
+
+
+@dataclass()
+class WriteStreamV2:
+    _process_batch: Callable[[Sequence[BatchItem]], None]
+    max_batch_size: int = 50_000
+    max_wait: timedelta = timedelta(seconds=1)
+    max_queue_size: int = 0  # Default to unlimited queue size
+    _item_queue: BackpressureQueue[BatchItem] = field(init=False)
+    _batch_queue: ReadQueue[Sequence[BatchItem]] | None = field(default=None)
+    _batch_thread: threading.Thread | None = field(default=None)
+    _process_thread: threading.Thread | None = field(default=None)
+    _executor: concurrent.futures.Executor | None = field(default=None)
+    backpressure_mode: BackpressureMode = field(default=BackpressureMode.BLOCK)
+
+    @classmethod
+    def create(
+        cls,
+        nominal_data_source_rid: str,
+        process_batch: Callable[[Sequence[BatchItem]], None],
+        max_batch_size: int = 50_000,
```

Comment:
same here

Reply:
prefer defaults primarily at user interfaces and not in internal code (unless you need multiple behaviors for diff functions)